smart-contracts/proto-build/src/main.rs

```diff
@@ -0,0 +1,285 @@
+//! This is based on the proto-compiler code in github.com/informalsystems/ibc-rs.
+
+use regex::Regex;
+use std::collections::BTreeMap;
+use std::io::Write;
+use std::{
+    ffi::{OsStr, OsString},
+    fs::{self, create_dir_all, File},
+    io,
+    path::{Path, PathBuf},
+    process,
+};
+use walkdir::WalkDir;
+
+const PROTO_BUILD_DIR: &str = "proto-build";
+const BUF_CONFIG_FILE: &str = "buf.yaml";
+const BUF_GEN_CONFIG_FILE: &str = "buf.quasar.gen.yaml";
+const PROTO_DIR: &str = "packages/quasar-std/src";
+const TMP_BUILD_DIR: &str = "/tmp/tmp-protobuf/";
+const QUASAR_DIR: &str = "../";
+
+macro_rules! info {
+    ($msg:expr) => {
+            println!("[info] {}", $msg)
+    };
+    ($fmt:expr, $($arg:tt)+) => {
+        info!(&format!($fmt, $($arg)+))
+    };
+}
+
+fn main() {
+    let tmp_build_dir: PathBuf = TMP_BUILD_DIR.parse().unwrap();
+    let proto_dir: PathBuf = PROTO_DIR.parse().unwrap();
+
+    if tmp_build_dir.exists() {
```

Comment:
Should we also cleanup this dir after running main?

Reply:
yes, done